pkg/agent/inteface.go

```diff
@@ -98,12 +98,19 @@ type AgentAddonOptions struct {
 	ConfigCheckEnabled bool
 }
 
-type CSRSignerFunc func(csr *certificatesv1.CertificateSigningRequest) []byte
+type CSRConfigurationsFunc func(cluster *clusterv1.ManagedCluster,
+	addon *addonapiv1alpha1.ManagedClusterAddOn) ([]addonapiv1alpha1.RegistrationConfig, error)
 
-type CSRApproveFunc func(cluster *clusterv1.ManagedCluster, addon *addonapiv1alpha1.ManagedClusterAddOn, csr *certificatesv1.CertificateSigningRequest) bool
+type CSRSignerFunc func(cluster *clusterv1.ManagedCluster,
```

Comment:
this will change the signature of the function, how do we resolve compatibility problem

Reply:
This is a breaking change; it is easy to migrate to the new version. The added cluster and addon parameters can be ignored if not needed. However, the developer still needs to make some slight changes.